libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/Binary/RoundTrip.hs

```diff
@@ -195,3 +208,55 @@ roundTripCoreEraTypesSpec = do
     roundTripShareEraTypeSpec @era @VState
     roundTripShareEraTypeSpec @era @CertState
     roundTripShareEraTypeSpec @era @UTxO
+
+data EraRuleProof era (rs :: [Symbol]) where
+  EraRuleProofEmpty :: EraRuleProof era '[]
+  EraRuleProofHead ::
+    ( Show (PredicateFailure (EraRule r era))
+    , Eq (PredicateFailure (EraRule r era))
+    , EncCBOR (PredicateFailure (EraRule r era))
+    , DecCBOR (PredicateFailure (EraRule r era))
+    , Arbitrary (PredicateFailure (EraRule r era))
```

Comment:
Let's switch it to `EraRuleFailure`. This will improve inference a bit and will test that we didn't mess up any `EraRuleFailure` instances
```suggestion
    ( Show (EraRuleFailure r era)
    , Eq (EraRuleFailure r era)
    , EncCBOR (EraRuleFailure r era)
    , DecCBOR (EraRuleFailure r era)
    , Arbitrary (EraRuleFailure r era)
    , EraRuleFailure r era ~ PredicateFailure (EraRule r era)
```